app/bettermail.go

```diff
@@ -24,33 +26,62 @@ func init() {
 	http.HandleFunc("/hook-test-harness", hookTestHarnessHandler)
 	http.HandleFunc("/test-mail-send", testMailSendHandler)
 	http.HandleFunc("/_ah/bounce", bounceHandler)
+	http.HandleFunc("/test-subject", testSubjectHandler)
+}
+
+type EmailThread struct {
+	CommitSHA string
+	Subject string
+}
+
+func createThread(sha string, subject string, c context.Context) {
+	thread := EmailThread {
+	    CommitSHA: sha,
+	    Subject: subject,
+	}
+	key := datastore.NewIncompleteKey(c, "EmailThread", nil)
```

Comment:
You can use `sha` as the key (`datastore.newKey(c, "EmailThread", sha, ...)`, and then you can use `datastore.Get` to more directly look it up.